src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -662,10 +662,17 @@ abstract class BTypes {
 
     def isInterface: Either[NoClassBTypeInfo, Boolean] = info.map(i => (i.flags & asm.Opcodes.ACC_INTERFACE) != 0)
 
-    def superClassesTransitive: Either[NoClassBTypeInfo, List[ClassBType]] = info.flatMap(i => i.superClass match {
-      case None => Right(Nil)
-      case Some(sc) =>  sc.superClassesTransitive.map(sc :: _)
-    })
+    def superClassesChain: Either[NoClassBTypeInfo, List[ClassBType]] = try {
```

Comment:
@lrytz Rather than recurring on the `superClassesChain`, which gives `Either[NoClassBTypeInfo, List[ClassBType]]`, I would extract a recursive `superClassesChain: List[ClassBType]`, without `Either` or `try`, and wrap only the initial call.